### PR TITLE
Support react-native-reanimated 2

### PR DESCRIPTION
**Pull request by pirroman**
https://github.com/computerjazz/react-native-draggable-flatlist/issues/247. There are no ts errors. All tests passed.

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -862,17 +861,17 @@ class DraggableFlatList<T> extends React.Component<
             ? styles.hoverComponentHorizontal
             : styles.hoverComponentVertical,
           {
-            opacity: this.hoverComponentOpacity,
+            opacity: this.hoverComponentOpacity[" __value"],
```
Comment (computerjazz) [inline src/index.tsx:864]: why is this necessary?

**Review by computerjazz — COMMENTED** [review]
looks good! just one question

File: src/index.tsx (hunk 1 above)
Comment (pirroman) [reply src/index.tsx]: It turns out it's not necessary. Reverted it, thank you!

**Comment by Jekiwijaya** [comment]
can't wait this to get merged, @pirroman @computerjazz  👍 👍 👍

**Comment by chrisvasey** [comment]
This is exciting, really excited to be able to use this package again!

**Comment by computerjazz** [comment]
Looks like a few conflicts arose due to a recent bugfix. Changes look good overall, I think we can merge once addressed.

That said I'm confused about the sentiment here -- the Reanimated 2 api is backwards compatible, and I'm currently running the DraggableFlatList example in an Expo project with Reanimated 2. To my knowledge, this PR will not fundamentally change anything, except maybe get rid of some warnings. Is there something else preventing you from using this package @Jekiwijaya @chrisvasey ?

**Comment by chrisvasey** [comment]
With v2 I am getting issues where sometimes child elements don't render when dragged/a bunch of errors. If I downgrade to v1 it fixes all the issues but sadly I need v2 for other parts of my app.

I wish I understood how the package worked to go into more detail but that has been my experience the last few days of trying different things 🙂

**Comment by computerjazz** [comment]
I see, thanks for the explanation, it sounds like reanimated v2 may have some regressions with the v1 api then. I wouldn't expect those to be fixed with these changes.

**Comment by pirroman** [comment]
@computerjazz I've had problems with react native cli with the release build(+hermes). After these edits, everything worked. Reanimated v2 has broken backward compatibility with RN below version 0.62. For example, the work with the refs has changed: before `const flatListNode = this.flatlistRef.current && this.flatlistRef.current.getNode();`
after `const flatListNode = this.flatlistRef.current;`

**Review by computerjazz — APPROVED** [review]
looks good, thanks!

**Comment by nelsonprsousa** [comment]
simulator_screenshot_B2D261E3-E6A7-4F89-96E7-FE0F073FCA41
Is this warning still expected while using 2.6.0? @computerjazz
